pipx/pipxrc.py

```diff
@@ -0,0 +1,193 @@
+import json
+import logging
+from pathlib import Path
+import re
+import textwrap
+from typing import List, Dict, NamedTuple, Any, Optional
+
+from pipx.Venv import PipxVenvMetadata, Venv
+from pipx.util import PipxError
+
+
+PIPX_INFO_FILENAME = "pipxrc.json"
+
+
+class JsonEncoderHandlesPath(json.JSONEncoder):
+    def default(self, obj):
+        # only handles what json.JSONEncoder doesn't understand by default
+        if isinstance(obj, Path):
+            return {"__type__": "Path", "__Path__": str(obj)}
+        return super().default(obj)
+
+
+def _json_decoder_object_hook(json_dict):
+    if json_dict.get("__type__", None) == "Path" and "__Path__" in json_dict:
+        return Path(json_dict["__Path__"])
+    return json_dict
+
+
+class PackageInfo(NamedTuple):
+    package_or_url: Optional[str]
```

Comment:
should the version be included here?

Reply:
The version of the package?  

I could probably do that.  I started by only putting the information in here that we get as arguments and need for arguments to re-play an install.  But I can add more.  I have to check, can I just get that info from the venv_metadata?

Reply:
I'm not sure. I have the suspicion it might be useful for various operations, or be needed in the future. If we can't find a need we don't need to include it.

Reply:
I included it already in the set of things inherited from VenvMetadata.  It's in there.